Bump to latest PG minors in CI images

hanefi opened the pull request:
This PR aims to capture the changes in our test suite to use the latest PG minor versions for 13 and 14.



Blocked on #6075 as we need to have matching libpq and PG minor versions in images here.

hanefi commented:
> Can we set min messages to error?

This worked like a charm. I did this for all failing failure tests and now we are fine.

All that is left to fix is `isolation_get_distributed_wait_queries_mx`. @onderkalaci will be working on a fix for this one.

hanefi commented:
A PR @JelteF merged today https://github.com/citusdata/citus/pull/6197 introduced a new problem after a rebase. Apparently, isolation tests started to use the session name in `application_name`. This caused the outputs to change after a PG minor bump. Instead of updating the tests, I decided to remove that column so that I do not force developers do upgrade their PG versions to latest minors.

```diff
 classid|  objid|objsubid|mode            |application_name                           |backend_type  |query                                       
 -------+-------+--------+----------------+-------------------------------------------+--------------+--------------------------------------------
-      0|8429800|       5|RowExclusiveLock|isolation/isolation_ref2ref_foreign_keys|client backend|     INSERT INTO ref_table_3 VALUES (7, 5);
-      0|8429801|       5|RowExclusiveLock|isolation/isolation_ref2ref_foreign_keys|client backend|     INSERT INTO ref_table_3 VALUES (7, 5);
-      0|8429802|       4|ShareLock       |isolation/isolation_ref2ref_foreign_keys|client backend|     INSERT INTO ref_table_3 VALUES (7, 5);
-      0|8429802|       5|RowExclusiveLock|isolation/isolation_ref2ref_foreign_keys|client backend|     INSERT INTO ref_table_3 VALUES (7, 5);
+      0|8429800|       5|RowExclusiveLock|isolation/isolation_ref2ref_foreign_keys/s2|client backend|     INSERT INTO ref_table_3 VALUES (7, 5); 
+      0|8429801|       5|RowExclusiveLock|isolation/isolation_ref2ref_foreign_keys/s2|client backend|     INSERT INTO ref_table_3 VALUES (7, 5); 
+      0|8429802|       4|ShareLock       |isolation/isolation_ref2ref_foreign_keys/s2|client backend|     INSERT INTO ref_table_3 VALUES (7, 5); 
+      0|8429802|       5|RowExclusiveLock|isolation/isolation_ref2ref_foreign_keys/s2|client backend|     INSERT INTO ref_table_3 VALUES (7, 5); 
 (4 rows)
```

onderkalaci commented:
> Apparently, isolation tests started to use the session name in application_nam

Yes, see https://github.com/citusdata/citus/pull/6210 for details

> I think it's fine to tell devs to have the latest PG versions if they want all tests to pass. So I would prefer if we keep the the application_name in the output, because it can help a lot with debugging (especially with the application name in the new output, because it's much more useful).


I think for our devs, it might be OK but we sometimes have users running tests and it is not great that the tests fails. Could we maybe add an explicit version test (&&comments) such that when it fails, it is clear that the rest might fail because of minor version incompatibility? Just thinking loudly, not sure if easily doable.

JelteF commented:
I think this query should work, to assert the correct minor version.
```
select (major = 13 AND minor >= 7) OR (major = 14 AND minor >= 4) OR (major >= 15) FROM (select split_part(version, '.', 1)::integer major, nullif(split_part(version, '.', 2), '')::integer minor from regexp_replace(split_part(version(), ' ', 2), 'beta|alpha|rc', '') version) q;
```

JelteF commented:
But TBH, I'm pretty sure our tests fail for older minor Postgres versions anyway. The isolation test output changed a while back for instance. So I'm not sure if it's worth it to introduce this query for the rare cases where users are running tests on old minor versions of postgres.

onderkalaci commented:
> But TBH, I'm pretty sure our tests fail for older minor Postgres versions anyway.

Yeah, agree. lets add 1 line comment to https://github.com/citusdata/citus/blob/main/CONTRIBUTING.md and tell that we *mostly* expect the users to use the latest minor Postgres versions for test stability?

This PR blocks PG 15 branch, so the priority is relatively high. So, @hanefi and @JelteF are you OK for adding 1 line/few lines comments and merge this?

hanefi commented:
Update: I spent some time trying to come up with a function that removes the session name from `application_name` if it exists. I was not successful.

I will now implement what you suggested above. Revert my changes to the test, add a comment, and update contributing.md

JelteF commented:
> I spent some time trying to come up with a function that removes the session name from application_name if it exists. I was not successful.

To be clear, I very much like the session name in there for debugging.

hanefi commented:
I updated the commits here, and added some comments and documentation.

- 9ec9209fd938fe2a19876a514ae6c851d53b6f92 bumps the PG version, adds the necessary documentation and comments.
- e33ba7da9e750824c9bd8dc3b3c137af3143e99b decreases `client_min_messages` to prevent some test failures.
- 616b1758c2b07154e74a6ae51fddf375f62d0172 adds more normalization rules to not have differing outputs across different minor versions. I expect this commit to be reverted as part of https://github.com/citusdata/citus/pull/5908
- a570dfead2b693b8548e8dbb51187d6f294283ca removes unused and misleading defaults in CI configs.
